SDC supporting multiple advertising sets (#471)
nrfconnect/sdk-nrfxlib: softdevice_controller/CHANGELOG.rst

Added
=====

* Added support for multiple advertising sets (DRGN-15426).
* Added radio front-end module (FEM) support, based on the :ref:`mpsl_fem` (DRGN-11059).
* Added support for the vendor-specific HCI command: Read Supported Vendor Specific Commands (DRGN-13763).
* Added support for the vendor-specific HCI command: Zephyr Read Key Hierarchy Roots (DRGN-13237).

Changes
=======

* Moved permanent limitations from the :ref:`nrf:known_issues` page to :ref:`softdevice_controller_limitations`.

> * Increassd the the maximum supported radio output power on nRF53 Series devices from 0 dBm to 3 dBm.

Reviewer: ```suggestion
* Increased the maximum supported radio output power on nRF53 Series devices from 0 dBm to 3 dBm.
```

  If the output power is above 0 dBm, NRF_VREQCTRL->VREGRADIO.VREQH is set (DRGN-15476).

Bug fixes
=========

* Fixed an issue where a slave connection could disconnect prematurely if there were scheduling conflicts with other roles (DRGN-15469).
* Fixed an issue where the channel map provided by the LE Host Set Channel Classification HCI command was not applied on the secondary advertising channels (DRGN-13594).
* The SoftDevice Controller can now be qualified on nRF52832 (DRGN-15382).
* Fixed an issue where setting a legacy advertiser's scan response data using extended advertising HCI commands corrupted the advertising data (DRGN-15465).
* Fixed an issue where, in rare cases, an assert could occur when receiving a packet as a slave.
  This could only occur after performing a data length procedure on Coded PHY (DRGN-15251).
* Fixed an issue where "HCI Read RSSI" would always return a Command Disallowed (0x0C) error code (DRGN-15310).

> * Fixed an issue where setting radio output power using the vendor-specific HCI command: Zephyr Write TX Power Level returned "Unsupported Feature or Parameter value (0x11).

Reviewer: ```suggestion
* Fixed an issue where setting radio output power using the vendor-specific HCI command Zephyr Write TX Power Level returned "Unsupported Feature or Parameter value (0x11)".
```

>   Now the controller will select an output power level that is lower or equal to the one requested. The command return success and the selected power level (DRGN-15369).

Reviewer: ```suggestion
  Now the controller will select an output power level that is lower or equal to the one requested. 
  The command returns success and the selected power level (DRGN-15369).
```

Reviewer: Line break + return**s**

> * Fixed an issue where an assert could occur when running an extended advertiser with maximum data length and minimum interval on Coded PHY. The assert would only occur if there were scheduling conflicts (DRGN-15694).

Reviewer: ```suggestion
* Fixed an issue where an assert could occur when running an extended advertiser with maximum data length and minimum interval on Coded PHY. 
  The assert would only occur if there were scheduling conflicts (DRGN-15694).
```

Reviewer: Line break.

Added
=====

* Added support for multiple advertising sets (DRGN-15426).
* Added radio front-end module (FEM) support, based on the :ref:`mpsl_fem` (DRGN-11059).
* Added support for the vendor specific HCI command: Read Supported Vendor Specific Commands (DRGN-13763).
* Added support for the vendor specific HCI command: Zephyr Read Key Hierarchy Roots (DRGN-13237).

> * On nRF53 Series devices, the maximum supported radio output power is increased from 0 dBm to 3 dBm.

Reviewer: Should this not go under Changes?

Reviewer: ```suggestion
* Increased the maximum supported radio output power on nRF53 Series devices from 0 dBm to 3 dBm.
```

Reviewer 2: - [ ] Move "Increased the maximum supported radio output power on nRF53 Series devices from 0 dBm to 3 dBm." to "Changes"

>   In this case, NRF_VREQCTRL->VREGRADIO.VREQH will be set (DRGN-15476).

Reviewer: What do we mean with "in this case"? If the maximum supported radio output power is increased?

Reviewer 2: If the output power is above 0 dBm, NRF_VREQCTRL->VREGRADIO.VREQH will be set

Reviewer: OK, then let's say that. :)
```suggestion
  If the output power is above 0 dBM, NRF_VREQCTRL->VREGRADIO.VREQH is set (DRGN-15476).
```

Bug fixes
=========

> * Fixed an issue where a slave connection could disconnect prematurely if there were scheduling conflicts with other roles (DRGN-15469).

Reviewer: Can we use peripheral instead of slave?

Reviewer 2: The controller terminology as defined by the Bluetooth specification uses "master" and "slave". The next specification release will change this terminology. Until then I would prefer to use "slave" as done in the rest of this document.

Reviewer: ok

* Fixed an issue where the channel map provided by the LE Host Set Channel Classification HCI command was not applied on the secondary advertising channels (DRGN-13594).
* The SoftDevice Controller can now be qualified on nRF52832 (DRGN-15382).
* Fixed an issue where setting a legacy advertiser's scan response data using extended advertising HCI commands corrupted the advertising data (DRGN-15465).
* Fixed an issue where, in rare cases, an assert could occur when receiving a packet as a slave.
  This could only occur after performing a data length procedure on Coded PHY (DRGN-15251).
* Fixed an issue where "HCI Read RSSI" would always return a Command Disallowed (0x0C) error code (DRGN-15310).

> * Fixed an issue where setting radio output power using the Vendor specific HCI command: Zephyr Write TX Power Level returned "Unsupported Feature or Parameter value (0x11).

Reviewer: ```suggestion
* Fixed an issue where setting radio output power using the vendor-specific HCI command Zephyr Write TX Power Level returned "Unsupported Feature or Parameter value (0x11)".
```

Reviewer 2: We use "Vendor specific" in many other places in this document. Should we update all occurrences to use "vendor-specific"?

Reviewer: Yes, please (unless it's part of the actual command name).

>   Now the controller will select an output power level that is lower or equal to the one requested. It will return success and the selected power level (DRGN-15369).

Reviewer: ```suggestion
  Now the controller will select an output power level that is lower or equal to the one requested. 
  The command will return success and the selected power level (DRGN-15369).
```

> * Fixed an issue where an assert could occur when running an extended advertiser with maximum data length and minimum interval on Coded PHY. The assert would only occur if there are scheduling conflicts (DRGN-15694).

Reviewer: ```suggestion
* Fixed an issue where an assert could occur when running an extended advertiser with maximum data length and minimum interval on Coded PHY. 
  The assert would only occur if there were scheduling conflicts (DRGN-15694).
```

  * ble_controller_rand_vector_get_blocking -> sdc_soc_rand_vector_get
  * ble_controller_ecb_block_encrypt -> sdc_soc_ecb_block_encrypt

> * vendor-specific HCI APIs have been renamed (DRGN-14701):

Reviewer: ```suggestion
* Vendor-specific HCI APIs have been renamed (DRGN-14701):
```
